core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -557,7 +561,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    */
   private def replay(
       eventLog: FileStatus,
-      bus: ReplayListenerBus): Option[FsApplicationAttemptInfo] = {
+      bus: ReplayListenerBus,
+      eventsFilter: (String) => Boolean = (s: String) => true): Option[FsApplicationAttemptInfo] = {
```

Comment:
No, I mean it looked like the method would never need to use any filter except for the start/end event one. So it doesn't need it as a parameter. That is, why is it up to the caller? the method filters in what it needs to produce the info it needs. I could be missing something.